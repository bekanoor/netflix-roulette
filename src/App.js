import P from './components/UiKit/typography/P';

export default function App() {
  return (
      <div className='wrapper'>
        <P font='large'>qsw</P>
      </div>
  );
}
