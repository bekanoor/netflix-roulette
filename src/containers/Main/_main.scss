.film-card {
  width: 320px;

  &__image-container {
    
  }
  &__image {
    width: 100%;
  }
  &__title-container {
    display: flex;
    justify-content: space-between;
  }
  &__release-date {
    color: white;
    font-family: 'Montserrat', sans-serif;
    font-size: 14px;
    width: 66px;
    height: 26px;
    border: 2px solid #979797;
    border-radius: 5px;
  }
}