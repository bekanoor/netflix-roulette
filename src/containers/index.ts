export * from './Header/'
export * from './Main/'
export * from './Footer'
export * from './MainPage' 
export * from  './ViewPage/'
export * from './FilmCard'
