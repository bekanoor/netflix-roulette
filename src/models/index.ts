import { moviesType, stateTypes, actionType } from './interfaces'

export type { moviesType, stateTypes, actionType };