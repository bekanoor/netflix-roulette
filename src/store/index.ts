export * from './actions'
export *  from './store'
export * from './moviesReducer'
export * from './searchParamReducer'
