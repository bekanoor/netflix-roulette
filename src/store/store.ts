import { reducer } from './reducer'

import { createStore } from 'redux'


const store = createStore(reducer)

export { store }

export type RootState = ReturnType<typeof store.getState >
export type AppDispatch = typeof store.dispatch  
