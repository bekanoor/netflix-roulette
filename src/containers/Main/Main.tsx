import { FilmCard } from '../'
import { getGenreOutput } from '../../utils'
import { moviesType } from '../../models'

interface IProps {
  movies: Array<moviesType>
}

const Main = (props: IProps) => {
  return (
    <main className='movies-wrapper' data-testid='main-test'>
      {props.movies.map((item) => {
        const { poster_path, title, release_date, id, genres } = item
        const genreResult = getGenreOutput(genres)

        return (
          <FilmCard
            id={id}
            key={id}
            cover={poster_path}
            filmTitle={title}
            releaseDate={release_date.substring(0, 4)}
            genre={genreResult}
          ></FilmCard>
        )
      })}
    </main>
  )
}

export default Main
