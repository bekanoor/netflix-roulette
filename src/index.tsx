import App from './App'
import { store } from './store'

import { render } from 'react-dom'
import { Provider } from 'react-redux'
import './style.scss'

const rootElement = document.getElementById('root')

render(
  <Provider store={store}>
    <App />
  </Provider>,
  rootElement
)
