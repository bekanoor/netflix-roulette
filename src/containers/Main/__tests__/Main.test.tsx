import { render } from '@testing-library/react'
import { moviesType } from 'src/models'
import Main from '../Main'
import '@testing-library/jest-dom'
import { Provider } from 'react-redux'
import { store } from '../../../store'

const data: Array<moviesType> = [
  {
    vote_average: 4.7,
    title: 'Kill Bill: Vol 2',
    genres: ['Oscar winning Movie'],
    release_date: '2004-12-12',
    poster_path: '../assets/images/Kill-Bill-Vol-2.jpg',
    runtime: 132,
    id: 2,
    overview:
      'The Bride continues her quest of vengeance against her former boss and lover Bill, the reclusive bouncer Budd, and the treacherous, one-eyed Elle.',
  },
]

describe('test main', () => {
  test('should render main correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Main movies={data} />
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })

  test('dom elements really exist', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Main movies={data} />
      </Provider>
    )
    expect(getByTestId('main-test')).toBeInTheDocument()
  })
})
