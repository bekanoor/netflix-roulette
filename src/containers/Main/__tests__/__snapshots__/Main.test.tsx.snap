// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test main should render main correctly 1`] = `
<div>
  <main
    class="movies-wrapper"
    data-testid="main-test"
  >
    <div
      class="film-card"
      data-testid="film-card-test"
      id="2"
    >
      <div
        class="film-card__image-container"
      >
        <img
          alt="image"
          class="film-card__image"
          src="../assets/images/Kill-Bill-Vol-2.jpg"
        />
      </div>
      <div
        class="film-card__title-container"
      >
        <p
          class="film-card__title"
        >
          Kill Bill: Vol 2
        </p>
        <button
          class="film-card__release-date"
        >
          2004
        </button>
      </div>
      <p
        class="film-card__genre"
      >
        Oscar winning Movie
      </p>
    </div>
  </main>
</div>
`;
